oracle/src/network/mod.rs

```diff
@@ -1,29 +1,42 @@
 use discovery::Discovery;
 use futures::FutureExt;
+use gossip::Gossip;
+use proto::{data::SignedTicker, ProtocolEvent};
 use reth_tracing::tracing::{error, info};
 use std::{
     future::Future,
     net::{IpAddr, Ipv4Addr, SocketAddr},
     pin::Pin,
     task::{Context, Poll},
 };
+use tokio::sync::broadcast::Sender;
 
 mod discovery;
+mod gossip;
 pub(crate) mod proto;
 
 /// The Network struct is a long running task that orchestrates discovery of new peers and network
 /// gossiping via the RLPx subprotocol.
 pub(crate) struct Network {
```

Comment:
let's rename this to something different than network, maybe OracleNetwork or similar?